src/prefect/engine.py

```diff
@@ -717,21 +734,39 @@ async def pause_flow_run(timeout: int = 300, poll_interval: int = 10):
             any configured flow-level timeout, the flow might fail even after resuming.
         poll_interval: The number of seconds between checking whether the flow has been
             resumed. Defaults to 10 seconds.
+        reschedule: Flag that will reschedule the flow run if resumed. Instead of
+            blocking execution, the flow will gracefully exit (with no result returned)
+            instead. To use this flag, a flow needs to have an associated deployment and
+            results need to be configured.
+        key: An optional key to prevent calling pauses more than once. This defaults to
+            the number of pauses observed by the flow so far, and prevents pauses that
+            use the "reschedule" option from running the same pause twice. A custom key
+            can be supplied for custom pausing behavior.
     """
-
     if TaskRunContext.get():
         raise RuntimeError("Cannot pause task runs.")
 
     frc = FlowRunContext.get()
     logger = get_run_logger(context=frc)
+    client = get_client()
+
+    pause_counter = _observed_flow_pauses(frc)
+    pause_key = key or str(pause_counter)
 
     logger.info("Pausing flow, execution will continue when this flow run is resumed.")
-    client = get_client()
     response = await client.set_flow_run_state(
         frc.flow_run.id,
-        Paused(),
+        Paused(timeout_seconds=timeout, reschedule=reschedule, pause_key=pause_key),
     )
 
+    if response.status == SetStateStatus.ABORT:
+        if response.details.reason == "This pause has already fired.":
+            return
```

Comment:
😬 Can we have this be a REJECT with RUNNING instead of an ABORT? I'd rather not perform logic based on the text in the reason and I think an ABORT should always signal an exit of the process.

Reply:
regarding this stuff, I was going to refine this in a future PR with a "CONTINUE" action

Reply:
this has been changed to be a reject, that matches much better to what we want to see happen and is distinct by more than a message